clang/lib/DPCT/MigrateCmakeScript.cpp

```diff
@@ -200,3 +321,279 @@ void migrateCmakeScriptOnly(
+      std::transform(Command.begin(), Command.end(), Command.begin(),
+                     [](unsigned char c) { return std::tolower(c); });
+
+      if (Command == "set") {
+        Index++; // Skip '('
+        std::string VarName;
+        std::string Value;
+
+        // Get the begin of firt argument of set
+        Index = skipWhileSpaces(Input, Index);
+        Begin = Index;
+
+        // Get the end of the first argument of set
+        Index = gotoEndOfCmakeWord(Input, Index, ')');
+        End = Index;
+
+        // Get the name of the first argument
+        VarName = Input.substr(Begin, End - Begin);
+
+        // Get the begin of the second argument
+        for (Index = End + 1; Index < Size && isWhitespace(Input[Index]);
+             Index++) {
+        }
+        Begin = Index;
+
+        // Get the end of the second argument
+        Index = gotoEndOfCmakeWord(Input, Begin + 1, ')');
+        End = Index;
+
+        // Get the name of the second argument
+        Value = Input.substr(Begin, End - Begin);
+
+        // Only check the set commands which has two arguments
+        if (Input[Index] == ')' &&
+            !llvm::StringRef(VarName).starts_with("CMAKE_") &&
+            !llvm::StringRef(VarName).starts_with("_CMAKE_ ")) {
+          VariablesMap[VarName] = Value;
+        }
+      }
+
+      // Go the ')' of cmake command
+      Index = gotoEndOfCmakeCommandStmt(Input, Index);
+    }
+
+    Index++;
+  }
+}
+
+static std::vector<std::string> split(const std::string &Input,
+                                      const std::string &Delimiter) {
+  std::vector<std::string> Vec;
+  if (!Input.empty()) {
+
+    size_t Index = 0;
+    size_t Pos = Input.find(Delimiter, Index);
+    while (Index < Input.size() && Pos != std::string::npos) {
+      Vec.push_back(Input.substr(Index, Pos - Index));
+
+      Index = Pos + Delimiter.size();
+      Pos = Input.find(Delimiter, Index);
+    }
+    // Append the remaining part
+    Vec.push_back(Input.substr(Index));
+  }
+  return Vec;
+}
+
+std::string getVarName(const std::string &Variable,
+                       const std::map<std::string, std::string> &VariablesMap) {
+  std::string Value;
+  if (Variable[0] == '$' && Variable[1] == '{') {
+    auto Name = Variable.substr(2, Variable.size() - 3);
+
+    auto Iter = VariablesMap.find(Name);
+    if (VariablesMap.find(Name) != VariablesMap.end()) {
+      Value = Iter->second;
+    }
+  } else {
+    Value = Variable;
+  }
+
+  return Value;
+}
+
+static std::string processArgOfCmakeVersionRequired(
+    const std::string &Arg,
+    const std::map<std::string, std::string> &VariablesMap) {
+  size_t Pos = Arg.find("...");
+  std::string ReplArg;
+  if (Pos != std::string::npos) {
+    const auto StrArray = split(Arg, "...");
+
+    std::string MinVer = getVarName(StrArray[0], VariablesMap);
+    std::string MaxVer = getVarName(StrArray[1], VariablesMap);
+
+    if (std::atof(MinVer.c_str()) >= 3.24) {
```

Comment:
Please fix the magic number in follow up PR.

Reply:
ok